bin/openness_report.py

```diff
@@ -0,0 +1,138 @@
+import tempfile
+import gzip
+import json
+import yaml
+from collections import defaultdict
+
+from functools import partial
+import traceback
+import unicodecsv
+import codecs
+
+proxy= os.environ['http_proxy']
+
+
+class Records():
+    def __init__(self, file, fmt_file):
+        self.file = file
+        if not os.path.isfile(self.file):
+            self.file = None
+        self.download_file = None
+        with open(fmt_file, 'r') as f:
+            presets = yaml.load(f)
+        fmt = None
+        for it in presets['presets']:
+            if it['preset_name']=='canada_resource_format':
+                fmt = it['values']['choices']
+        if not fmt:
+            raise Exception('no resource format loaded')
+        self.fmt = filter(lambda x: 'openness_score' in x, fmt)
+
+    def __delete__(self):
+        if not self.file:
+            if self.download_file:
+                os.unlink(self.download_file)
+                print('temp file deleted', self.download_file)
+
+    def download(self):
+        if not self.file:
+            # dataset http://open.canada.ca/data/en/dataset/c4c5c7f1-bfa6-4ff6-b4a0-c164cb2060f7
+            url='http://open.canada.ca/static/od-do-canada.jl.gz'
+            r = requests.get(url, stream=True)
+
+            f = tempfile.NamedTemporaryFile(delete=False)
+            for chunk in r.iter_content(1024 * 64):
+                    f.write(chunk)
+            f.close()
+            self.download_file = f.name
+
+        records = []
+        fname = self.file or f.name
+        try:
+            with gzip.open(fname, 'rb') as fd:
+                for line in fd:
+                    records.append(json.loads(line.decode('utf-8')))
+                    if len(records) >= 50:
+                        yield (records)
+                        records = []
+            if len(records) >0:
+                yield (records)
+        except GeneratorExit:
+            pass
+        except:
+            import traceback
+            traceback.print_exc()
+            print('error reading downloaded file')
+
+
+    def iter_resources(self):
+        count = 0
+        reports = defaultdict(lambda: defaultdict(int))
+        for records in self.download():
+            for record in records:
+                report = reports[record['organization']['title']]
+                formats = map(lambda x: x['format'], record['resources'])
+                scores = map(lambda x:x['openness_score'],
+                             filter(lambda x: x['value'] in formats,
+                                    self.fmt))
+                scores.append(1)
+                score = max(scores)
+
+                for r in record['resources']:
+                    if 'data_includes_uris' in r.get('data_quality', []):
+                        score = max(4, score)
+                        if 'data_includes_links' in r.get('data_quality', []):
+                            score = max(5, score)
+                report[score] += 1
```

Comment:
if it's a python3 import issue we could move the logic to a module that is importable from either python2 or 3: this could be the start of a good thing :-)